File: samples/java/app/src/main/java/org/remdev/wetweexapp/generator/GeneratorViewModel.java
package org.remdev.wetweexapp.generator;

import org.remdev.wetweex.live.AnyThreadMutableLiveData;
import org.remdev.wetweex.live.TweexMutableLiveData;
import org.remdev.wetweex.viewmodel.BaseTweexProgressViewModel;

public class GeneratorViewModel extends BaseTweexProgressViewModel {

    private TweexMutableLiveData<String> generatedText = new AnyThreadMutableLiveData<>();

    public TweexMutableLiveData<String> getGeneratedText() {
        return generatedText;
    }
}

File: samples/kotlin/app/src/main/java/org/remdev/wetweexapp/generator/GeneratorViewModel.kt
package org.remdev.wetweexapp.generator

import org.remdev.wetweex.live.AnyThreadMutableLiveData
import org.remdev.wetweex.live.TweexMutableLiveData
import org.remdev.wetweex.viewmodel.BaseTweexProgressViewModel

open class GeneratorViewModel : BaseTweexProgressViewModel() {

    open val generatedText: TweexMutableLiveData<String> = AnyThreadMutableLiveData<String>()
}